crates/nostr-relay-pool/src/relay/internal.rs

```diff
@@ -1060,11 +1061,7 @@ impl InternalRelay {
             msgs.push(ClientMessage::event(event));
         }
 
-        // Batch send messages
-        self.batch_msg(msgs, opts).await?;
-
-        // Hanlde responses
-        time::timeout(Some(opts.timeout), async {
+        let handle_responses = time::timeout(Some(opts.timeout), async {
             let mut published: HashSet<EventId> = HashSet::new();
             let mut not_published: HashMap<EventId, String> = HashMap::new();
             let mut notifications = self.internal_notification_sender.subscribe();
```

Comment:
Maybe is enough to move the `let mut notifications = self.internal_notification_sender.subscribe();` outside the `time::timeout`, before the `self.batch_msg(msgs, opts).await?;`. Should be cleaner as solution.

What do you think?